### PR TITLE
feat: add contained-list

### PR DESCRIPTION
**Pull request by maicongodinho**
New Contained List component 🥳

Reference: https://carbondesignsystem.com/components/contained-list/usage/
React team implementation: https://react.carbondesignsystem.com/?path=/story/components-containedlist--default

### CODE REVIEW

**Comment by maicongodinho** [comment]
I have read the DCO document and I hereby sign the DCO.

File: src/contained-list/contained-list-item.component.ts
Comment (Akshat55) [reply src/contained-list/contained-list-item.component.ts:21]: We can have a call and discuss this if you'd like 😄

**Review by Akshat55 — CHANGES_REQUESTED** [review]

File: src/icon/icon.module.ts
Comment (maicongodinho) [reply src/icon/icon.module.ts]: This 47c3462 would be a better approach?

File: src/contained-list/contained-list-size.enum.ts
Comment (maicongodinho) [reply src/contained-list/contained-list-size.enum.ts]: How about this e5529fe?

File: src/contained-list/contained-list-kind.enum.ts
Comment (maicongodinho) [reply src/contained-list/contained-list-kind.enum.ts]: How about this e5529fe?

File: src/contained-list/contained-list-item.component.ts
Comment (maicongodinho) [reply src/contained-list/contained-list-item.component.ts]: This should do the trick [ebc5f54](https://github.com/carbon-design-system/carbon-components-angular/pull/2464/commits/ebc5f549a9a3eec1341d710ceb0cc8d336733500)

File: src/contained-list/contained-list-item.component.ts
Comment (maicongodinho) [reply src/contained-list/contained-list-item.component.ts]: I think this resolve the problem 2c259b182cc483082d1b25af6b66d014d6aaaa91

File: src/contained-list/contained-list-item.component.ts
Comment (Akshat55) [inline src/contained-list/contained-list-item.component.ts:62]: What if the icon is not available in carbon? I think it's best to accept template ref here 🤔

File: src/contained-list/contained-list-item.component.spec.ts
Comment (Akshat55) [reply src/contained-list/contained-list-item.component.spec.ts:24]: At the end of the day, this is unordered list. We should just do simple checks, to see if appropriate classes are assigned based on input.

* For example, check to see if `ibm-contained-list` is assigned to the component (Parent).
* Next, update the size & check if the appropriate class is assigned to the the component. Alternatively, you can check to see if the value of size is the one you have set.
* Change the kind & check if appropriate css classes are assigned (Or check to see if the value in the component has change).

**Review by Akshat55 — CHANGES_REQUESTED** [review]

File: src/contained-list/contained-list-item.component.ts
Comment (maicongodinho) [reply src/contained-list/contained-list-item.component.ts]: Done here 80228e5

File: src/contained-list/contained-list.component.spec.ts
Comment (maicongodinho) [reply src/contained-list/contained-list.component.spec.ts]: Check if this is OK 102a0f8

File: src/contained-list/contained-list-item.component.spec.ts
Comment (maicongodinho) [reply src/contained-list/contained-list-item.component.spec.ts]: Check if this is OK 102a0f8

File: src/contained-list/contained-list-item.component.ts
Comment (Akshat55) [inline src/contained-list/contained-list-item.component.ts:62]:
```suggestion
	/**
	 * Provide an optional icon to render in front of the item's content.
	 */
	@Input() icon: TemplateRef<any> | string;
```

I thought about your original implementation with string @maicongodinho and I think we can make use of both! It's a really good idea in my opinion. What are your thoughts @zvonimirfras?

It would make it easier for teams that want to use ONLY carbon icons in their contained list. This would also save them the trouble of writing a `ng-template` 🤔

Comment (Akshat55) [reply src/contained-list/contained-list-item.component.ts:62]: If we do decide to go with this, we should add a comment letting users know that they need to register their `CARBON` icons.

File: src/contained-list/contained-list.stories.ts
Comment (Akshat55) [inline src/contained-list/contained-list.stories.ts:333]: I don't think we need this, React refers to stories with knobs as `playground`. We automatically add knobs using compodoc.


```suggestion
```

File: src/contained-list/contained-list.stories.ts
Comment (Akshat55) [inline src/contained-list/contained-list.stories.ts:270]:
```suggestion
export const withInteractiveItemsAndActions = withInteractiveItemsAndActionsTemplate.bind({});
```

File: src/contained-list/contained-list.stories.ts
Comment (Akshat55) [inline src/contained-list/contained-list.stories.ts:268]: Could you also add an expandable search component to this? React has it in their [storybook](https://react.carbondesignsystem.com/?path=/story/components-containedlist--with-interactive-items-and-actions), I THINK it should work, but I just want to double check.

File: src/contained-list/contained-list-story.module.ts
Comment (Akshat55) [inline src/contained-list/contained-list-story.module.ts:31]: Could you move this file under a `stories` directory. Have a look at the table component for reference

Would be something like:
```
.
└── stories/
    └── contained-list-story.module.ts
```

Also, do we need to import & register the size 20 icons? I think we are using only size 16.

**Review by Akshat55 — CHANGES_REQUESTED** [review]

File: src/contained-list/contained-list.stories.ts
Comment (maicongodinho) [reply src/contained-list/contained-list.stories.ts:268]: At the time I open the PR the new docs on search was incomplete, I will verify now and retry the implementation.

File: src/contained-list/contained-list-item.component.ts
Comment (maicongodinho) [reply src/contained-list/contained-list-item.component.ts:62]: Yeah I agree, what is your thoughts on that @zvonimirfras?

File: src/contained-list/contained-list-story.module.ts
Comment (maicongodinho) [reply src/contained-list/contained-list-story.module.ts]: Done on [cdc3821](https://github.com/carbon-design-system/carbon-components-angular/pull/2464/commits/cdc38211338c179aba4f3ff630c434de4f6ca46a)

File: src/contained-list/contained-list.component.ts
Comment (maicongodinho) [reply src/contained-list/contained-list.component.ts]: Check this [e63c0bd](https://github.com/carbon-design-system/carbon-components-angular/pull/2464/commits/e63c0bdbed9881c41eb66c4c96b979294c5af1c2)

File: src/contained-list/contained-list-item.component.ts
Comment (maicongodinho) [reply src/contained-list/contained-list-item.component.ts]: Check this [e63c0bd](https://github.com/carbon-design-system/carbon-components-angular/pull/2464/commits/e63c0bdbed9881c41eb66c4c96b979294c5af1c2)

File: src/contained-list/contained-list-item.component.ts
Comment (zvonimirfras) [reply src/contained-list/contained-list-item.component.ts:62]: Let's do it!

Comment (maicongodinho) [reply src/contained-list/contained-list-item.component.ts]: Check that @Akshat55 [43b7f43](https://github.com/carbon-design-system/carbon-components-angular/pull/2464/commits/43b7f432965b9cd5112d60e5ad00ce2c2008173f)

File: src/contained-list/contained-list.stories.ts
Comment (maicongodinho) [reply src/contained-list/contained-list.stories.ts]: I  think we are having some scss problems with search. Look at e86c763 to see how the component is placed on the story.

**Review by Akshat55 — DISMISSED** [review]
AWESOME WORK @maicongodinho! 🎉🎉

It's ready to go on my end! Will have a final review with Z sometime this week & possible merge!

File: src/contained-list/contained-list.stories.ts
Comment (Akshat55) [reply src/contained-list/contained-list.stories.ts]: It's expected, we can't do much about it. We would have to add `style: block` to the `cds-search` tag. Overall, it isn't a major issue & can be tackled by the users!

**Comment by Akshat55** [comment]
Just an FYI @maicongodinho  We are looking to merge this into master soon - Once `@carbon/styles` package is updated to account for angular components (including display property to classes) it will be merged. It is good to go on your end.

**Comment by maicongodinho** [comment]
As @Akshat55 asked, I changed the PR target branch to master (before was next) and update the PR branch with the latest master changes.

File: src/contained-list/contained-list-item.component.ts
Comment (Akshat55) [inline src/contained-list/contained-list-item.component.ts:41]:
```suggestion
```

We no longer needed, latest styles package will take care of this.

File: src/contained-list/contained-list.component.ts
Comment (Akshat55) [inline src/contained-list/contained-list.component.ts:125]:
```suggestion
```

We don't need host binding since it is redundant of ngClass, I prefer using ngClass w/ div here.

File: src/contained-list/contained-list.component.ts
Comment (Akshat55) [inline src/contained-list/contained-list.component.ts:43]:
```suggestion
```

Don't need this since we will be assigning the classes to the div instead of the host.

**Review by Akshat55 — CHANGES_REQUESTED** [review]
Need to remove some redundant code like styles & HostBinding classes.

Could you also export the module you created in `src/index.ts`. Otherwise the module may not be exported in library 😅.

**Comment by maicongodinho** [comment]
Thank you for the heads up. I think everything is good now if you want to check.

**Review by zvonimirfras — CHANGES_REQUESTED** [review]

File: src/contained-list/contained-list.component.ts
Comment (Akshat55) [reply src/contained-list/contained-list.component.ts:73]:
```suggestion
	@Input() labelId = `contained-list-${ContainedList.count++}-header`;
```

**Review by Akshat55 — APPROVED** [review]

**Review by zvonimirfras — APPROVED** [review]

**Pull request merged**
